### PR TITLE
VACMS-12351: Home page v2 hub list menu

### PR DESCRIPTION
**Pull request by dsasser**
## Description
This PR moves to using a Drupal menu for controlling the _new_ home page hub list links. This builds on the Drupal work that adds a new menu for managing the links. The original home page is using Benefit Hub Landing Page nodes, but we are moving to using a menu for the new home page to allow for non-BHLP links to be included while maintaining canonical management in the CMS.

This PR depends on the Drupal menu work: https://github.com/department-of-veterans-affairs/va.gov-cms/issues/11968

relates to https://github.com/department-of-veterans-affairs/va.gov-cms/issues/12351  

## Testing done & Screenshots
Screen Shot 2023-02-06 at 1 08 46 PM

### Testing will be performed on the Tugboat instance created by the related BE PR.
CMS: https://cms-7kaomxlwh9vfmuyg1u9xup0etcqzmggi.demo.cms.va.gov/
WEB: https://web-7kaomxlwh9vfmuyg1u9xup0etcqzmggi.demo.cms.va.gov/

1. Visit the new nome page at https://web-7kaomxlwh9vfmuyg1u9xup0etcqzmggi.demo.cms.va.gov/new-home-page/new-home-page
   - [ ] Validate that the home page hub links appear as [designed in Sketch](https://www.sketch.com/s/3aa40506-4be2-46cc-876b-93f1a9f3a857/p/EAD305BC-3082-4F81-ADEC-C2486F7B6BEF/canvas).
2. Visit the Drupal CMS menu at https://cms-7kaomxlwh9vfmuyg1u9xup0etcqzmggi.demo.cms.va.gov/admin/structure/menu/manage/home-page-hub-list/admin/structure/menu/manage/home-page-hub-list 
3. Then change a link to be disabled and save the form
4. Re-arrange the items and save the form
5. Edit the "Health care" node at /node/67/edit and set the Editorial workflow to "Archived" and save the form
6. Do a [content release](https://cms-7kaomxlwh9vfmuyg1u9xup0etcqzmggi.demo.cms.va.gov/admin/content/deploy) using the branch `VACMS-12351-home-page-v2-hub-list-menu` (screenshot below)
7. Wait for the content release status to be 'Ready'. This will take a while. It may take up to one minute for the status of new content releases to be reflected here, so refresh the page after a minute to see if the release is underway.
8. Visit the new home page at https://cms-pfcgaquu99xi4wzksftxmvukusulz3bh.ci.cms.va.gov/new-home-page
   - [ ] Validate the disabled link no longer appears on the new home page
   - [ ] Validate the order of the menu items is reflected
   - [ ] Validate any links to unpublished nodes no longer appear on the new home page

### Content release screenshot
Screen Shot 2023-02-07 at 9 01 46 AM

## Definition of done
- [x] Events are logged appropriately
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [X] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by apisandipas — APPROVED** [review]

File: src/site/stages/build/drupal/home.js [hunk 1]
```diff
@@ -85,6 +86,22 @@ function addHomeContent(contentData, files, metalsmith, buildOptions) {
...
+    const homePreviewHubs = homePageHubListMenuQuery.links.filter(link => {
+      // Any disabled links should not be displayed.
+      if (!link.enabled) {
+        return false;
+      }
+      // If the link has a linkedEntity, and the linkedEntity is not published, it
+      // should not be displayed.
+      return (
+        !link.entity.linkedEntity ||
+        (link.entity.linkedEntity && link.entity.linkedEntity.entityPublished)
```
Comment (jtmst) [inline src/site/stages/build/drupal/home.js:101]: Comment says that 'and the linkedEntity is not published',  should this be `!link.entity.linkedEntity.entityPublished`?

Comment (dsasser) [reply src/site/stages/build/drupal/home.js:101]: The comments don't help clarify what is happening I suspect, but we are effectively returning the value of `link.entity.linkedEntity.entityPublished` if link.entity.linkedEntity is set. So when  `link.entity.linkedEntity.entityPublished` = `false`, we want to remove it from the page, and therefore return false to the filter() method.

For instance, given the current link is:
```json
      {
        "enabled": true,
        // etc...
        "entity": {
          "linkedEntity": {
            "entityPublished": false,
            // etc...
          }
        }
      },
```

line 100
`!link.entity.linkedEntity` equates to `false`

line 101
`link.entity.linkedEntity` equates to `true`
`link.entity.linkedEntity.entityPublished` equates to `false`

false || false = false

If we flip the entityStatus to `true` this becomes...

line 100
`!link.entity.linkedEntity` equates to `false`

line 101
`link.entity.linkedEntity` equates to `true`
`link.entity.linkedEntity.entityPublished` equates to `true`

false || true = true

Hope that makes sense.

**Review by jtmst — APPROVED** [review]

File: src/site/stages/build/drupal/home.js (hunk 1 above)
Comment (jtmst) [reply src/site/stages/build/drupal/home.js:101]: gotcha, thanks for clarifying.

**Comment by laflannery** [comment]
A couple questions/comments:

1. The FE display of the Hubs is not exactly the same as the Sketch file, the width of the columns on the site is wider than sketch. Is this important? You can see this based on which words wraps in the descriptions in the screenshots below
**Website:**
image
**Sketch:**
image

**Comment by dsasser** [comment]
> 1. It's not clear to me how I would go about adding another icon choice into the dropdown, for future use. Is this something that a manager of this Hub Menu would need to be able to do?

The menu will be managed by admins only at this time, who can add new icon options if needed by manipulating the correct field on the menu. However, since that also requires a code change on the front end (in content-build), it needs to be ticketed and implemented by PW. A future need may warrant changing that behavior, but since these items change rarely, we didn't see a need to change it at this time.

> The FE display of the Hubs is not exactly the same as the Sketch file, the width of the columns on the site is wider than sketch. Is this important? You can see this based on which words wraps in the descriptions in the screenshots below.

I can't speak to why the Sketch file is slightly off from the actual site, but no changes were made on the frontend to affect the width of the columns. @thejordanwood has signed off on the design, so I think we can ignore that unless she has some additional context to add.

**Comment by laflannery** [comment]
2 more questions just to cover all the bases:
- When an editor choose an existing landing page to add to this menu it is NOT supposed to pull in these fields from that node correct, because we are eventually removing those fields? I believe this is correct but I just want to confirm
image
- Is it possible to add the same existing node to the menu twice? What I mean is that I tried to change the Health care page to the same link the disability page is linked to (VA Disability Compensation (68)) but it wouldn't show in the dropdown options, is that on purpose?
image

**Comment by dsasser** [comment]
> * When an editor choose an existing landing page to add to this menu it is NOT supposed to pull in these fields from that node correct, because we are eventually removing those fields? I believe this is correct but I just want to confirm

That is correct, it is not supposed to pull the data from the BHLP node. Some of those fields will eventually be removed from the BHLP.

> Is it possible to add the same existing node to the menu twice? What I mean is that I tried to change the Health care page to the same link the disability page is linked to (VA Disability Compensation (68)) but it wouldn't show in the dropdown options, is that on purpose? I don't really see a need for why I would want to do this but I just want to be thorough

It is possible to add the same node to the menu. The problem you are seeing is that there is a limit of 10 results returned for autocomplete results, and the node you wanted to add was not in the result set (there are at least 11 nodes which include 'va disability compensation' in the title). This is a limitation with the way autocomplete is handled across the site. You can, however, paste the full node title + node id in the field and it will save. eg: `VA disability compensation (68)`. I have made that change to the health care link that can be verified here: https://cms-7kaomxlwh9vfmuyg1u9xup0etcqzmggi.demo.cms.va.gov/admin/structure/menu/item/11735/edit?destination=/admin/structure/menu/manage/home-page-hub-list.

**Review by laflannery — APPROVED** [review]

**Pull request merged**